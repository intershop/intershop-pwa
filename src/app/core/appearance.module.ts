import { NgModule } from '@angular/core';
import { NgbPopoverConfig } from '@ng-bootstrap/ng-bootstrap';
import { ToastrModule } from 'ngx-toastr';

import { IconModule } from './icon.module';
import { ThemeService } from './utils/theme/theme.service';

@NgModule({
  imports: [
    IconModule,
    ToastrModule.forRoot({
      closeButton: true,
      timeOut: 3000,
      positionClass: 'toast-top-full-width', // toast-top-center
      preventDuplicates: true,
    }),
  ],
})
export class AppearanceModule {
  constructor(popoverConfig: NgbPopoverConfig, themeService: ThemeService) {
    popoverConfig.placement = 'top';
    popoverConfig.triggers = 'hover';
    popoverConfig.container = 'body';

    themeService.init();
  }
}
