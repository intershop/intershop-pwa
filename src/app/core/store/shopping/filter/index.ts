// API to access ngrx filter state
export * from './filter.actions';
export * from './filter.selectors';
