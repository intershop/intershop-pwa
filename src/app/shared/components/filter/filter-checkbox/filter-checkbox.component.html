<!--- TODO: implement Show All-->
<ul class="filter-list">
  <ng-container *ngFor="let facet of getFacets()">
    <li class="filter-item filter-layer{{ facet.level }}" [ngClass]="{ 'filter-selected': facet.selected }">
      <div>
        <label class="filter-item-checkbox-label-native">
          <input type="checkbox" [checked]="facet.selected" (change)="filter(facet)" />
          <span class="filter-item-name"> {{ facet.displayName }} </span>
          <span class="count"> ({{ facet.count }}) </span>
        </label>
      </div>
    </li>
  </ng-container>
  <li
    class="filter-item"
    *ngIf="filterElement.limitCount !== -1 && filterElement.facets.length > filterElement.limitCount"
  >
    <a class="btn-link" (click)="showAll = !showAll">{{
      (showAll ? 'search.filter.show_less.link' : 'search.filter.show_all.link') | translate
    }}</a>
  </li>
</ul>
