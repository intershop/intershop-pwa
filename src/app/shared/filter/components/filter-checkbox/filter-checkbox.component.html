<!--- TODO: implement Show All-->
<ul class="filter-list">
  <ng-container *ngFor="let facet of filterElement.facets">
    <li class="filter-item filter-layer{{ facet.level }}" [ngClass]="{ 'filter-selected': facet.selected }">
      <div>
        <label class="filter-item-checkbox-label-native">
          <input type="checkbox" [checked]="facet.selected" (change)="filter(facet)" />
          <span class="filter-item-name"> {{ facet.displayName }} </span>
          <span class="count"> ({{ facet.count }}) </span>
        </label>
      </div>
    </li>
  </ng-container>
</ul>
