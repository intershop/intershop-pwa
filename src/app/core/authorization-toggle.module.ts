import { ModuleWithProviders, NgModule } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import { map } from 'rxjs/operators';

import { AuthorizationToggleDirective } from './directives/authorization-toggle.directive';
import { AuthorizationToggleService, checkPermission } from './utils/authorization-toggle/authorization-toggle.service';
import { whenTruthy } from './utils/operators';

@NgModule({
  declarations: [AuthorizationToggleDirective],
  exports: [AuthorizationToggleDirective],
})
export class AuthorizationToggleModule {
  private static permissions = new ReplaySubject<string[]>(1);

  static forTesting(...permissions: string[]): ModuleWithProviders<AuthorizationToggleModule> {
    AuthorizationToggleModule.switchTestingPermissions(...permissions);
    return {
      ngModule: AuthorizationToggleModule,
      providers: [
        {
          provide: AuthorizationToggleService,
          useValue: {
            isAuthorizedTo: (permission: string) =>
              AuthorizationToggleModule.permissions.pipe(
                whenTruthy(),
                map(perms => checkPermission(perms, permission))
              ),
          },
        },
      ],
    };
  }

  static switchTestingPermissions(...permissions: string[]) {
    AuthorizationToggleModule.permissions.next(permissions);
  }
}

export { AuthorizationToggleService } from './utils/authorization-toggle/authorization-toggle.service';
export { AuthorizationToggleGuard } from './guards/authorization-toggle.guard';
