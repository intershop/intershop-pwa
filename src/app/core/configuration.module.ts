import { NgModule } from '@angular/core';

import { SPECIAL_HTTP_ERROR_HANDLER } from './interceptors/icm-error-mapper.interceptor';
import { createPaymentErrorHandler } from './utils/http-error/create-payment.error-handler';
import { editPasswordErrorHandler } from './utils/http-error/edit-password.error-handler';
import { LoginUserErrorHandler } from './utils/http-error/login-user.error-handler';
import { requestReminderErrorHandler } from './utils/http-error/request-reminder.error-handler';
import { updatePasswordErrorHandler } from './utils/http-error/update-password.error-handler';

@NgModule({
  providers: [
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: updatePasswordErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useClass: LoginUserErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: requestReminderErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: editPasswordErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: createPaymentErrorHandler, multi: true },
  ],
})
export class ConfigurationModule {}
