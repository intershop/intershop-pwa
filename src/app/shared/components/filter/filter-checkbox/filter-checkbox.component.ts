import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';

import { Facet } from 'ish-core/models/facet/facet.model';
import { Filter } from 'ish-core/models/filter/filter.model';
import { URLFormParams } from 'ish-core/utils/url-form-params';

/**
 * The Filter Checkbox Component displays a filter group. The items of the filter group are presented as checkboxes.
 *
 * @example
 * <ish-filter-checkbox
 *               [filterElement]="element"
 *               (applyFilter)="applyFilter($event)"
 * </ish-filter-checkbox>
 */
@Component({
  selector: 'ish-filter-checkbox',
  templateUrl: './filter-checkbox.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class FilterCheckboxComponent {
  @Input() filterElement: Filter;
  @Output() applyFilter: EventEmitter<{ searchParameter: URLFormParams }> = new EventEmitter();

  /**
   * two-way-binding (banana in a box) [(showAll)]="showAllElements[element.name]"
   */
  @Output()
  showAllChange = new EventEmitter<boolean>();
  private showAllValue = false;
  @Input()
  get showAll() {
    return this.showAllValue;
  }
  set showAll(val) {
    this.showAllValue = val;
    this.showAllChange.emit(this.showAllValue);
  }

  filter(facet: Facet) {
    this.applyFilter.emit({ searchParameter: facet.searchParameter });
  }

  /**
   * sort selected to top, increase limitCount to selectedCount on showLess
   */
  getFacets() {
    const facets = [...this.filterElement.facets];

    const selectedFacetsCount = facets.filter(x => x.selected).length;

    return this.showAll || this.filterElement.limitCount === -1
      ? facets
      : facets
          .sort((a, b) => (a.selected > b.selected ? -1 : a.selected < b.selected ? 1 : 0))
          .slice(0, Math.max(this.filterElement.limitCount || 0, selectedFacetsCount));
  }
}
