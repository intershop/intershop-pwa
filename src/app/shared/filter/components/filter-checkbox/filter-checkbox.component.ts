import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';

import { Facet } from 'ish-core/models/facet/facet.model';
import { Filter } from 'ish-core/models/filter/filter.model';

/**
 * The Filter Checkbox Component displays a filter group. The items of the filter group are presented as checkboxes.
 *
 * @example
 * <ish-filter-checkbox
 *               [filterElement]="element"
 *               (applyFilter)="applyFilter($event)"
 * </ish-filter-checkbox>
 */
@Component({
  selector: 'ish-filter-checkbox',
  templateUrl: './filter-checkbox.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class FilterCheckboxComponent {
  @Input() filterElement: Filter;
  @Output() applyFilter: EventEmitter<{ searchParameter: string }> = new EventEmitter();

  filter(facet: Facet) {
    this.applyFilter.emit({ searchParameter: facet.searchParameter });
  }
}
