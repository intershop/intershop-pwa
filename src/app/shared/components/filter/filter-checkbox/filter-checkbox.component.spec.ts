import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { FaIconComponent } from '@fortawesome/angular-fontawesome';
import { NgbCollapse } from '@ng-bootstrap/ng-bootstrap';
import { TranslateModule } from '@ngx-translate/core';
import { MockComponent } from 'ng-mocks';

import { Filter } from 'ish-core/models/filter/filter.model';
import { SanitizePipe } from 'ish-core/pipes/sanitize.pipe';

import { FilterCheckboxComponent } from './filter-checkbox.component';

describe('Filter Checkbox Component', () => {
  let component: FilterCheckboxComponent;
  let fixture: ComponentFixture<FilterCheckboxComponent>;
  let element: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [TranslateModule.forRoot()],
      declarations: [FilterCheckboxComponent, MockComponent(FaIconComponent), MockComponent(NgbCollapse), SanitizePipe],
    }).compileComponents();
  }));

  beforeEach(() => {
    const filterElement = {
      name: 'Brands',
      limitCount: -1,
      facets: [
        { name: 'AsusName', count: 4, displayName: 'Asus' },
        { name: 'LogitechName', count: 5, displayName: 'Logitech', selected: true },
      ],
    } as Filter;
    fixture = TestBed.createComponent(FilterCheckboxComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    component.filterElement = filterElement;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
    expect(element).toMatchSnapshot();
  });
});
