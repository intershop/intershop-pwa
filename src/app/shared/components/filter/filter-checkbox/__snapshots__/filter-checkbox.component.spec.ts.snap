// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Checkbox Component should be created 1`] = `
<ul class="filter-list">
  <li class="filter-item filter-layer">
    <div>
      <label class="filter-item-checkbox-label-native"
        ><input type="checkbox" /><span class="filter-item-name"> Asus </span
        ><span class="count"> (4) </span></label
      >
    </div>
  </li>
  <li class="filter-item filter-layer filter-selected">
    <div>
      <label class="filter-item-checkbox-label-native"
        ><input type="checkbox" /><span class="filter-item-name"> Logitech </span
        ><span class="count"> (5) </span></label
      >
    </div>
  </li>
</ul>
`;
